hack/jenkins/e2e.sh

```diff
@@ -472,6 +472,21 @@ case ${JOB_NAME} in
     : ${ENABLE_DEPLOYMENTS:=true}
     ;;
 
+  kubernetes-e2e-gke-subnet)
+    : ${DOGFOOD_GCLOUD:="true"}
+    : ${E2E_CLUSTER_NAME:="jkns-gke-subnet"}
+    : ${E2E_NETWORK:="auto-subnet"}
```

Comment:
It's straightforward https://cloud.google.com/compute/docs/subnetworks#networks_and_subnetworks : in the CLI or UI, just check the "auto" mode.  No further options.